### DISCUSSION TITLE
[PSD10] Exercise 2.44

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#169] [Esercizi (A-L)]
![image](https://user-images.githubusercontent.com/69389461/210061094-95a23dee-a97c-4fe6-945e-5d40a274a1be.png)

![image](https://user-images.githubusercontent.com/69389461/210061274-4fa9e5dd-0eb7-40fd-9350-55ce19683e4b.png)

![image](https://user-images.githubusercontent.com/69389461/210061056-0ebbb939-b841-42f3-8326-c9678a3a2b6f.png)

**Comment by Elia-Belli** [comment]
Per calcolare il propagation e contamination delay di un circuito si deve valutare il delay del percorso critico, che solitamente è quello che presenta più porte logiche: in questo caso ci sono 2 percosi con lo stesso numero di porte $A \rightarrow Y$ e $G \rightarrow Y$, quindi confrontiamo i loro delay e prendiamo il maggiore.

- $A \rightarrow Y$: 
  **propagation delay**: $(30 + 20* 2 + 30) = 100ps$
  **contamination delay**: $(25 + 2* 15 + 25) = 80ps$
- $G \rightarrow Y$: 
  **propagation delay**: $(30 + 30* 2 + 20) = 110ps$
  **contamination delay**: $(25 + 25 * 2 + 15) = 85ps$

Quindi il percorso critico è $G \rightarrow Y$, il **contamination delay** del circuito è $85ps$ e il  **propagation delay** del circuito è $110ps$.

**Comment by fraxb04** [comment]
La logica è giusta, ed i risultati del circuito non semplificato sono corretti.
Tuttavia non capisco perché non si debba utilizzare il bubble pushing, ad esempio io ho fatto così:
![image](https://user-images.githubusercontent.com/122573871/213474071-74ec4e8d-03a9-43ae-944b-d508aefeaa85.png)

**Comment by Elia-Belli** [comment]
@fraxb04  Il Bubble Pushing è utile per passare dalla logica invertente (NAND e NOR) più efficiente per la macchina, a logica non-invertente (AND e OR) più comprensibile all'uomo.
Un circuito (non la funzione logica che rappresenta!) è univocamente identificato dalle porte logiche che lo costituiscono, quindi applicando il Bubble Pushing cambiano le porte e di conseguenza il circuito e il suo delay.

Detto questo l'esercizio chiede di ricavare i delay di **quel circuito** e non della sua versione con logica non-invertente.

**Comment by fraxb04** [comment]
ok grz, mille!!
